textproc/pandoc/Portfile

```diff
@@ -27,14 +27,21 @@ homepage            https://pandoc.org
 
 installs_libs       no
 
-checksums           rmd160  941a905dafbc79f959c218ff65508100b5fe15a2 \
-                    sha256  709629f264455f73da7afbcae1138e97cb375a7563cf9c251c059f54bb777259 \
-                    size    7651793
+checksums           rmd160  1c465118e3118f2534607feda25f11db94576966 \
+                    sha256  8f922a35eff1a96be323e496ba0ebe8ad985db17a84d2c553d672346913bea9c \
+                    size    7655959
 
-# See https://trac.macports.org/ticket/48971
-notes-append       "For PDF support, please install the texlive-latex and texlive-fonts-recommended packages."
+# https://github.com/jgm/pandoc/blob/main/tools/latex-package-dependencies.lua
+# comm -23 <(pandoc lua ${worksrcpath}/tools/latex-package-dependencies.lua | xargs -I{} find ${prefix}/share/texmf-texlive -name {} | sort -u | xargs -I{} find {} -depth 1 -type f | xargs port provides | awk '{print $NF}' | sort -u) <(port echo rdepof:texlive-xetex rdepends:texlive-fonts-recommended | grep texlive | sort -u)
+depends_run-append  port:groff \
+                    port:latexmk \
+                    port:texlive-bin-extra \
+                    port:texlive-context \
+                    port:texlive-latex-extra \
+                    port:texlive-luatex \
+                    port:texlive-xetex
```

Comment:
these are rather heavy dependencies... So dar they haven't appeared to be needed for the typical use for pandoc dependents. Possibly this should be put in a variant instead?

Reply:
maybe think about a  mactex variant to prevent double installations: https://ports.macports.org/search/?selected_facets=variants_exact:mactex

Reply:
> these are rather heavy dependencies... So dar they haven't appeared to be needed for the typical use for pandoc dependents. Possibly this should be put in a variant instead?

Yes, good point.

Default `pandoc` should include the basic capability to create pdf's, which comes from [`texlive-xetex`](https://github.com/jgm/pandoc/blob/main/doc/faqs.md). The footprint isn't much—a dozen photos or so—compared to `pandoc`'s 16 MB footprint.

The full dependencies (that I presume are used infrequently for various edge cases) can be added as a variant along with a corresponding note.

```sh
du -sm /opt/local/var/macports/software/texlive-{xetex,basic,bin,common,latex-recommended,plain-generic,luatex,context,latex-extra,bin-extra}
12	/opt/local/var/macports/software/texlive-xetex
23	/opt/local/var/macports/software/texlive-basic
17	/opt/local/var/macports/software/texlive-bin
1	/opt/local/var/macports/software/texlive-common
64	/opt/local/var/macports/software/texlive-latex-recommended
32	/opt/local/var/macports/software/texlive-plain-generic
25	/opt/local/var/macports/software/texlive-luatex
107	/opt/local/var/macports/software/texlive-context
695	/opt/local/var/macports/software/texlive-latex-extra
134	/opt/local/var/macports/software/texlive-bin-extra
```

Reply:
> they haven't appeared to be needed for the typical use for pandoc dependents

The heaviest use of `pandoc` comes through `nbconvert`, and I see that we don't specify this runtime dependency. I'll also add that to `py-nbconvert`.

https://github.com/jupyter/nbconvert/blob/5b836027e43fc29d607d73ec689c2efc1419a903/nbconvert/utils/pandoc.py#L130-L131

Reply:
> maybe think about a mactex variant to prevent double installations: https://ports.macports.org/search/?selected_facets=variants_exact:mactex

There's already a heavy dependency on `texlive-*` through the GHC build chain. If you know how to use the lua file of LaTeX dependencies commented in the Portfile, please add this as a separate PR.